src/ApiClient.php

```
@@ -69,9 +85,17 @@ public function search(string $searchTerm, string $locale = 'en-US', int $limit
             .'&region=US&quotesCount='.$limit
             .'&quotesQueryId=tss_match_phrase_query&multiQuoteQueryId=multi_quote_single_token_query&enableCb=false&enableNavLinks=true&enableCulturalAssets=true&enableNews=false&enableResearchReports=false&enableLists=false&listsCount=0&recommendCount=0&enablePrivateCompany=true';
 
-        $responseBody = (string) $this->client->request('GET', $url, ['headers' => $this->getHeaders()])->getBody();
+        $response = $this->client->request('GET', $url, ['headers' => $this->getHeaders(), 'cookies' => $this->getCookies()]);
+
+        $status = $response->getStatusCode();
+        if (200 !== $status) {
+            $this->registerClient();
```

Why are you calling `registerClient` when you're throwing an exception immediately after?

---

cookies are valid approx 24-48 hours only if any of it expired or invalid it will request new one

---

It doesn't explain why you're executing a request, when you're immediately afterwards ending the code execution with an exception. Then you could throw the exception right away, instead of waiting for another request.

---

ohh sorry, i noticed i forgot to remove this code